core/src/main/java/hudson/model/Computer.java

```diff
@@ -1109,6 +1109,11 @@ public String getSearchUrl() {
         return getUrl();
     }
 
+    @Override
+    public String getSearchIcon() {
+        return this.getIconClassName();
+    }
```

Comment:
seems unneeded? it implements the interface already

Reply:
It implements `IComputer` which doesn't extend `IconSpec`, could extend `IconSpec` however to fix it.

Without `getSearchIcon()`:
<img width="272" alt="Screenshot 2024-12-11 at 20 49 30" src="https://github.com/user-attachments/assets/0c568ccd-e88a-4a05-a0d1-14fca5016fb6" />

With `getSearchIcon()`:
<img width="296" alt="image" src="https://github.com/user-attachments/assets/c7d006ed-f8d9-4ee1-9e65-96e379b402d8" />

Reply:
Yeah just extend it in `IComputer`